tools/python/influx-migration/influx_migration.py

```diff
@@ -170,30 +230,44 @@ def bucket_create_rollback(host, token, bucket_name, org, skip_verify):
         client.close()
     return True
 
-def bucket_exists(host, token, bucket_name, skip_verify=False, org=None):
+def bucket_exists(host, token, bucket_name, skip_verify=False, org_name=None):
     """
     Checks for the existence of a bucket.
 
     :param str host: The host for the InfluxDB instance.
     :param str token: The token to use for verification.
     :param str bucket_name: The name of the bucket to verify.
     :param bool skip_verify: Whether to skip TLS certificate verification.
-    :param org: The name of the org to use for bucket verification
-    :type org: str or None
+    :param org_name: The name of the org to use for bucket verification
+    :type org_name: str or None
     :returns: Whether the bucket exists in the instance.
     :rtype: bool
     """
     try:
-        client = InfluxDBClient(url=host,
-            token=token, timeout=MILLISECOND_TIMEOUT, verify_ssl=not skip_verify, org=org)
-        if client.buckets_api().find_bucket_by_name(bucket_name) is None:
-            return False
+        with InfluxDBClient(url=host, token=token, timeout=MILLISECOND_TIMEOUT,
+            verify_ssl=not skip_verify) as client:
+            org_list = []
+            if org_name is not None:
+                client.org = org_name
+                org_list = client.organizations_api().find_organizations(org=org_name)
+                if len(org_list) <= 0:
+                    logging.debug(f"Org {org_name} could not be found when checking whether bucket {bucket_name} exists")
+                    return False
+                logging.debug(f"{len(org_list)} orgs matched with name {org_name}")
+            bucket = client.buckets_api().find_bucket_by_name(bucket_name)
+            if bucket is None:
+                return False
```

Comment:
Do we want to add a debug log stating the bucket doesn't exist?

Reply:
Sure, I will add a debug log for each return value.

Reply:
Added.